svm/src/account_loader.rs

```diff
@@ -167,35 +186,90 @@ pub(crate) fn load_accounts<CB: TransactionProcessingCallback>(
-            }
-            (_, Err(e)) => Err(e),
+        .map(|(transaction, validation_result)| {
+            load_transaction(
+                callbacks,
+                transaction,
+                validation_result,
+                error_metrics,
+                account_overrides,
+                feature_set,
+                rent_collector,
+                loaded_programs,
+            )
         })
         .collect()
 }
 
+fn load_transaction<CB: TransactionProcessingCallback>(
+    callbacks: &CB,
+    message: &impl SVMMessage,
+    validation_result: TransactionValidationResult,
+    error_metrics: &mut TransactionErrorMetrics,
+    account_overrides: Option<&AccountOverrides>,
+    feature_set: &FeatureSet,
+    rent_collector: &RentCollector,
+    loaded_programs: &ProgramCacheForTxBatch,
+) -> TransactionLoadResult {
+    match validation_result {
+        Err(e) => TransactionLoadResult::NotLoaded(e),
+        Ok(tx_details) => {
+            let load_result = load_transaction_accounts(
+                callbacks,
+                message,
+                tx_details.fee_payer_account,
+                tx_details.fee_payer_rent_debit,
+                &tx_details.compute_budget_limits,
+                error_metrics,
+                account_overrides,
+                feature_set,
+                rent_collector,
+                loaded_programs,
+            );
+
+            match load_result {
+                Ok(loaded_tx_accounts) => TransactionLoadResult::Loaded(LoadedTransaction {
+                    accounts: loaded_tx_accounts.accounts,
+                    program_indices: loaded_tx_accounts.program_indices,
+                    fee_details: tx_details.fee_details,
+                    rent: loaded_tx_accounts.rent,
+                    rent_debits: loaded_tx_accounts.rent_debits,
+                    rollback_accounts: tx_details.rollback_accounts,
+                    compute_budget_limits: tx_details.compute_budget_limits,
+                    loaded_accounts_data_size: loaded_tx_accounts.loaded_accounts_data_size,
+                }),
+                Err(err) => TransactionLoadResult::FeesOnly(FeesOnlyTransaction {
+                    load_error: err,
+                    fee_details: tx_details.fee_details,
+                    rollback_accounts: tx_details.rollback_accounts,
+                }),
+            }
+        }
+    }
+}
+
+#[derive(PartialEq, Eq, Debug, Clone)]
+struct LoadedTransactionAccounts {
+    pub accounts: Vec<TransactionAccount>,
+    pub program_indices: TransactionProgramIndices,
+    pub rent: TransactionRent,
+    pub rent_debits: RentDebits,
+    pub loaded_accounts_data_size: u32,
+}
+
+#[allow(clippy::too_many_arguments)]
 fn load_transaction_accounts<CB: TransactionProcessingCallback>(
     callbacks: &CB,
     message: &impl SVMMessage,
-    tx_details: ValidatedTransactionDetails,
+    fee_payer_account: AccountSharedData,
+    fee_payer_rent_debit: u64,
```

Comment:
didn't you just add the `ValidatedTransactionDetails` recently?
why change it back?

Reply:
Because I don't want to pass the entire struct by value anymore because if loading fails, we still need to recover the `fee_details` and `rollback_accounts` fields from that struct for processing fees-only transactions